### PR TITLE
tests: show just the last log as part of the debug output when check journal logs

### PR DESCRIPTION
**Pull request by sergiocazzolato**
The idea of this change is to avoid showing all the logs in the debug
output. The change makes visible just the last log in case there is no
match.

This is an example of the big log produced: https://api.travis-ci.org/v3/job/569278195/log.txt

### CODE REVIEW

File: tests/lib/journalctl.sh [hunk 1]
```diff
@@ -35,14 +35,20 @@ check_journalctl_ready(){
 check_journalctl_log(){
     expression=$1
     shift
-    for _ in $(seq 10); do
-        log=$(get_journalctl_log "$@")
-        if echo "$log" | grep -q -E "$expression"; then
+    # Initial checks dont print the log in the debug output
+    for iter in $(seq 9); do
```
Comment (stolowski) [inline tests/lib/journalctl.sh]: Is there any reason for the change from `for _ in $(seq 10)` to `iter` and 9?

File: tests/lib/journalctl.sh [hunk 2]
```diff
@@ -35,14 +35,20 @@ check_journalctl_ready(){
...
+        if get_journalctl_log "$@" | grep -q -E "$expression"; then
```
Comment (stolowski) [inline tests/lib/journalctl.sh]: Silly question, so `echo "$log" | grep -q ....` makes the output appear in the log, but `get_journalctl_log "$@" | grep -q ...` doesn't?

File: tests/lib/journalctl.sh (hunk 1 above)
Comment (sergiocazzolato) [reply tests/lib/journalctl.sh]: good catch, forgot to revert that :)

File: tests/lib/journalctl.sh (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/journalctl.sh]: This makes the log appears in the debug output.
    log=$(get_journalctl_log "$@")
So the idea is to make 
    get_journalctl_log "$@" | grep -q -E "$expression" 
that pipes the log to the grep command instead

Comment (zyga) [reply tests/lib/journalctl.sh]: I think Pawel's question is valid: both commands pipe to `grep -q` so why one logs and the other does not?
It seems to suggest `get_journalctl_log` is broken and that it writes to stderr.

A quick look at `get_journalctl_log` as well as `get_journalctl_log_from_cursor` doesn't show anything immediately broken. I will see what's going on here first.

Comment (zyga) [reply tests/lib/journalctl.sh]: I had a look at this and proposed a quick sanity check PR: https://github.com/snapcore/snapd/pull/7225 -- let's see if that passes. It removes the whole loop entirely and simplifies surrounding code.

**Comment by pedronis** [comment]
Latest discussions is that we want to pursue a different approach, @chipaca proposed to have ways to temporarely (in the tests that require this) reconfigure the snapd service to send stderr to a file and then grep that. And look at journal only when strictly needed. Closing this.

**Pull request closed without merging**